kernel/userspace.c

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <kernel.h>
+#include <string.h>
+#include <misc/printk.h>
+#include <kernel_structs.h>
+#include <sys_io.h>
+
+/**
+ * Kernel object validation function
+ *
+ * Retrieve metadata for a kernel object. This function is implemented in
+ * the gperf script footer, see gen_kobject_list.py
+ *
+ * @param obj Address of kernel object to get metadata
+ * @return Kernel object's metadata, or NULL if the parameter wasn't the
+ * memory address of a kernel object
+ */
+__weak _GENERIC_SECTION(.kobject_data.text)
+struct _k_object *_k_object_find(void *obj)
+{
+	/* This is the __weak stub implementation, the version created by
+	 * gperf will replace it in build phase 2
```

Comment:
FWIW: there's no need for this thing to actually be a function if it will never be called.  Linker names are typeless.  And no need to put it in a specific section if all you want is for the link to not fail.  Just a:

    __weak int _k_object_find;

...should be enough, no?  Or alternativesly a PROVIDES() directive in the linker script that is #if'd appropriately so it shows up only in the early linkage.

Reply:
i'll give that a try!